packages/apollo-link-error/README.md

```diff
@@ -29,13 +29,50 @@ Error Link takes a function that is called in the event of an error. This functi
 * `response`: The result returned from lower down in the link chain
 * `graphQLErrors`: An array of errors from the GraphQL endpoint
 * `networkError`: Any error during the link execution or server response, that wasn't delivered as part of the `errors` field in the GraphQL result
+* `forward`: A reference to the next link in the chain. Calling `return forward(operation)` in the callback will retry the request, returning a new observable for the upstream link to subscribe to. 
+
+Returns: `Observable<FetchResult> | void` The error callback can optionally return an observable from calling `forward(operation)` if it wants to retry the request. It should not return anything else. 
 
 <h2 id="error-categories">Error categorization</h2>
 
 An error is passed as a `networkError` if a link further down the chain called the `error` callback on the observable. In most cases, `graphQLErrors` is the `errors` field of the result from the last `next` call.
 
 A `networkError` can contain additional fields, such as a GraphQL object in the case of a [failing HTTP status code](http.html#Errors) from [`apollo-link-http`](http.html). In this situation, `graphQLErrors` is an alias for `networkError.result.errors` if the property exists.
 
+<h2 id="error-categories">Retrying failed requests</h2>
+
+An error handler might want to do more than just logging errors. You can check for a certain failure condition or error code, and retry the request if rectifying the error is possible. For example, when using some form of token based authentication, there is a need to handle re-authentication when the token expires. Here is an example of how to do this using `foward()`.
+```js
+onError(({ graphQLErrors, networkError, operation, forward }) => {
+    if (graphQLErrors) {
+      for (let err of graphQLErrors) {
+        switch (err.extensions.code) {
+          case 'UNAUTHENTICATED': 
+            // error code is set to UNAUTHENTICATED 
+            // when AuthenticationError thrown in resolver
+            
+            // modify the operation context with a new token
+            const oldHeaders = operation.getContext().headers;
+            operation.setContext({
+              headers: {
+                ...oldHeaders,
+                authorization: getNewToken(),
+              },
+            });
+            // retry the request, returning the new observable
+            return forward(operation);
+        }
+      }
+    }
+    if (networkError) {
+      console.log(`[Network error]: ${networkError}`);
+      // no need to retry if there was a network error
```

Comment:
Adding a sentence to this comment about using apollo-link-retry to deal with `networkError`s would help users understand the distinction between link-error and link-retry

Reply:
fixed